doc/content/reference/api/end_device.md

```diff
@@ -4,59 +4,114 @@ description: ""
 weight: 7
 ---
 
+End devices are registered in multiple registries. The Identity Server has a registry with end device metadata, the Network Server's registry contains the MAC configuration and MAC state (including network session keys), the Application Server keeps payload formatters and application session keys, the Join Server keeps the root keys.
+
+When registering end devices, we recommend registering them in the following order:
+
+- `EndDeviceRegistry.Create` (Identity Server)
+- `JsEndDeviceRegistry.Set` (Join Server, only for OTAA devices)
+- `NsEndDeviceRegistry.Set` (Network Server)
+- `AsEndDeviceRegistry.Set` (Application Server)
+
+When deleting end devices, we recommend deleting them in the reverse order
+
 ## The `EndDeviceRegistry` service
 
-{{< proto/method service="EndDeviceRegistry" method="Create" >}}
+The Identity Server's `EndDeviceRegistry` is the first place to register an end device. This registry stores the following [EndDevice fields](#message:EndDevice):
 
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
+- `ids` (with subfields)
+- `name`
+- `description`
+- `attributes`
+- `version_ids` (with subfields)
+- `network_server_address`
+- `application_server_address`
+- `join_server_address` (only for OTAA devices)
+- `service_profile_id`
+- `locations`
+- `picture`
 
-{{< proto/method service="EndDeviceRegistry" method="Get" >}}
+{{< proto/method service="EndDeviceRegistry" method="Create" >}}
 
-> TODO: [Document what fields to get from the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
+{{< proto/method service="EndDeviceRegistry" method="Get" >}}
 
 {{< proto/method service="EndDeviceRegistry" method="List" >}}
 
-> TODO: [Document what fields to get from the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Update" >}}
 
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Delete" >}}
 
 ## The `JsEndDeviceRegistry` service
 
-{{< proto/method service="JsEndDeviceRegistry" method="Set" >}}
+OTAA devices are registered in the Join Server's `JsEndDeviceRegistry`. This registry stores the following [EndDevice fields](#message:EndDevice):
```

Comment:
The OTAA part is (and should be) already described above, maybe better to just be consistent with other components?
```suggestion
The Join Server's `JsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
```

Reply:
I don't want to assume that all readers read all information at the top before they skip to here.